docs/src/references/agent/device-profile.md

```diff
@@ -0,0 +1,327 @@
+on_success = { status = "failed", reason = "Device profile application failed" }
+on_error = { status = "failed", reason = "Rollback failed" }
+
+#
+# End states
+#
+[successful]
+action = "cleanup"
+
+[failed]
+action = "cleanup"
+```
+
+* The workflow just proceeds to the `scheduled` state from the `init` state
+* The order of operation execution must be finalized before the `executing` state
+  and the `scheduled` state is an ideal candidate for that.
+  If the `builtin` action is specified in this state, the default order as defined in the input is used.
+  This order can be overridden by the user using a `script` action, if desired.
+  The script is expected to return the updated `operations` list which replaces the original list
+  and then proceed to the `executing` state.
+* The mandatory `executing` state simply passes the input to the `next_operation`.
+* The `next_operation` state chooses the next operation to be executed from the list of `operations` in the input.
+  When the built-in `iterator` action is used, the next operation is picked up sequentially from the `operations` list.
+  The target operation is captured into a `@next_operation` object in the payload,
+  with an `index` field representing the index position of that operation in the `operations` list, 
+  along with that operation's `operation` and `payload` values as follows:
+
+  ```json
+  {
+    "@next_operation": {
+      "index": 0,
+      "operation": "firmware_update",
+      "payload": {
+        "name": "core-image-tedge-rauc",
+        "remoteUrl": "https://abc.com/some/firmware/url",
+        "version": "20240430.1139"
+      }
+    },
+    ... // Other fields in the incoming payload
+  }
+  ```
+
+  If the `@next_operation` field is not present in the input payload.
+  one is added with an initial `index` value of `0` and the corresponding `operation` and `payload` values.
+  If the field already exists, the `index` value is incremented along with its `operation` and `payload` values.
+  Once the next operation is successfully computed, the workflow moves to the `on_next` target.
+  Once the `operations` list is exhausted (`index` value higher than its size),
+  the profile application is deemed complete and the workflow proceeds to the `on_success` target.
+  If the operation computation fails for some reason, then the workflow moves to the `on_error` target.
+  This builtin iteration logic can be overridden using a `script` action which can manipulate the order in any manner, dynamically.
+* The `apply_sub_operation` state executes the sub-operation defined in the `@next_operation` field in the payload.
+  The `input` to the sub-operation is also extracted from the `payload` field of the `@next_operation`.
+  As soon as the sub-operation is triggered, the workflow moves to the `awaiting_sub_operation` state defined as the `on_exec` target.
+* In the `awaiting_sub_operation` state, workflow just waits monitoring the state of the sub-operation completion.
```

Comment:
Rename the `awaiting_sub_operation` state to `awaiting_operation` (for consistency) - Also in the example workflow toml.